File: src/dataset/cpp/grains_exercism-felbit/gemini/GeminiFlash_grains_v1.cpp
#include "grains.h"

namespace grains
{

    ull square(unsigned short _square)
    {
        if (_square == 0 || _square > 64) {
            throw std::domain_error("square must be between 1 and 64");
        }
        return 1ULL << (_square - 1);
    }

    ull total()
    {
        return (1ULL << 64) - 1;
    }

}

File: src/dataset/typescript/grains_exercism-wackerow/gemini/GeminiFlash_grains_v1.ts
export default class Grains {
  public static square(digit: number): number {
    if (digit <= 0 || digit > 64 || !Number.isInteger(digit)) {
      throw 'Chessboard location must be integer from 1 to 64.';
    }
    return 2 ** (digit - 1);
  }

  public static total(): bigint {
    return 2n ** 64n - 1n;
  }
}